adapters/repos/db/roaringset/compactor.go

```diff
@@ -82,13 +151,41 @@ type Compactor struct {
 func NewCompactor(w io.WriteSeeker,
 	left, right *SegmentCursor, level uint16,
 	scratchSpacePath string, cleanupDeletions bool,
-	enableChecksumValidation bool,
+	enableChecksumValidation bool, maxNewFileSize int64,
 ) *Compactor {
+	// The layout of the segment is
+	//  header
+	//  data
+	//  check-sum
+	//
+	// However, it is challenging to calculate the length of the data (which is part of the header) before writing the
+	// file:
+	//
+	// big files (overhead is not that relevant)
+	// - write empty header
+	// - write data
+	// - seek back to start
+	// - write real header
+	// - seek to original position (after data)
+	// - write checksum
+	//
+	// For small files we use a dummy writer, that writes everything to memory before actually writing the file
+
+	var writer compactorWriter
+	var mw *memoryWriter
+	if maxNewFileSize < 4096 {
```

Comment:
this threshold could be a declared constant or configurable

Reply:
decided against it, because we are allocating memory anyways